source/Irrlicht/COpenGLDriver.h

```diff
@@ -846,6 +848,38 @@ namespace video
             typedef std::pair<COpenGLVAOSpec::HashAttribs,COpenGLVAO*> HashVAOPair;
             HashVAOPair                 CurrentVAO;
             core::vector<HashVAOPair>    VAOMap;
+            // TODO this has to be peridically searched for shaders with refcount=1 and cleared

```

Comment:
tell me what d you think about this

Reply:
why do you need this cache in the first place?

make a `removeFromCache(COpenGLSpecializedShader)` and `clearCache()`

Reply:
@Crisspl ?

Reply:
this look completely different now

Reply:
how doo you remove stuff from caches that are not VAOs ?

Reply:
COpenGLSpecializedShader has its object-local cache for <ctxID,GLname> as we talked few days before, and theres no other than vao caches in COpenGLDriver/SAuxCtx created by me. There will be cache for pipelines though ofc.. And deletion will look same as it lloks for vaos, so in `COpenGLDriver::cleanUpContextBeforeDelete()`

Reply:
will you also clear the pipeline cache same as VAO when you get too much in cache?

also what about framebuffer cache?

Reply:
@Crisspl status?

Reply:
framebuffer cache totally not important right now since it just works, todo when things will be working
>will you also clear the pipeline cache same as VAO when you get too much in cache?

not sure if it can be done the same if pipeline cache is going to be map (not vector)

Reply:
well the vector is sorted by hash, so the LRU removal is unfortunately a linear search XD

Not sure you could do any worse with a map

Reply:
@Crisspl TODO and relevant to the question of how you're caching pipelines.